docs/virtualview.md

```diff
@@ -0,0 +1,169 @@
+:::
+
+---
+
+## Props
+
+### `children`
+
+Content to render inside this `VirtualView`.
+
+| Type                     |
+| ------------------------ |
+| [React Node](react-node) |
+
+---
+
+### `onModeChange`
+
+Invoked when the `VirtualView` changes how it renders its children.
+
+If a callback is supplied, it may be invoked from different threads and priorities depending on the internal state change. This can be detected by checking the `mode` property on the event:
+
+- If `mode` is [`VirtualViewMode.Visible`](#virtualviewmode), the callback is being invoked from the main thread with immediate priority.
+- If `mode` is [`VirtualViewMode.Prerender`](#virtualviewmode) or [`VirtualViewMode.Hidden`](#virtualviewmode), the callback is being invoked from a background thread with transition priority.
+
+The callback will never be invoked consecutively with the same `mode` value. However, there are few guarantees about sequencing of events. Also, the callback may never be invoked with [`VirtualViewMode.Visible`](#virtualviewmode) even if it becomes visible, if the children were successfully prerendered.
+
+| Type                                               |
+| -------------------------------------------------- |
+| `md ([ModeChangeEvent](#modechangeevent)) => void` |
+
+---
+
+### `nativeID`
+
+An identifier for locating this view from native classes.
+
+| Type   |
+| ------ |
+| string |
+
+---
+
+### `style`
+
+| Type                           |
+| ------------------------------ |
+| [View Style](view-style-props) |
+
+---
+
+## Type Definitions
+
+### `ModeChangeEvent`
+
+Argument supplied to [`onModeChange`](#onmodechange).
+
+| Type   |
+| ------ |
+| object |
+
+**Properties:**
+
+| Name          | Type                                | Description                                                                                       |
+| ------------- | ----------------------------------- | ------------------------------------------------------------------------------------------------- |
+| mode          | [VirtualViewMode](#virtualviewmode) | New mode of the `VirtualView`.                                                                    |
+| target        | element                             | `VirtualView` emitting this event.                                                                |
+| targetRect    | [Rect](rect)                        | Layout of `target` relative to the nearest ancestor `ScrollView`.                                 |
+| thresholdRect | [Rect](rect)                        | Layout of the threshold that triggered this event, relative to the nearest ancestor `ScrollView`. |
+
+:::note
+For example, if a `VirtualView` enters the visible region of a [`ScrollView`](scrollview)...
+
+- `mode` would be [`VirtualViewMode.Visible`](#virtualviewmode)
+- `thresholdRect` would describe the current scroll position of the nearest ancestor [`ScrollView`](scrollview)
```

Comment:
> describe the current scroll position 

I'm wondering if we should word it as the visible viewport of the nearest ancestor ScrollView to be clearer what we mean by scroll position